tgui/packages/tgui/interfaces/ColorPickerModal.tsx

```diff
@@ -0,0 +1,766 @@
+        <Interactive
+          style={{ background }}
+          onMove={handleMove}
+          onKey={handleKey}
+          aria-label="Saturation"
+          aria-valuenow={Math.round(color.s)}
+          aria-valuemax={100}
+          aria-valuemin={0}
+        >
+          <Pointer
+            className="react-colorful__saturation-pointer"
+            left={color.s / 100}
+            color={hsvaToHslString({
+              h: color.h,
+              s: color.s,
+              v: color.v,
+              a: 1,
+            })}
+          />
+        </Interactive>
+      </div>
+    );
+  },
+);
+
+interface ValueProps {
+  className?: string;
+  color: HsvaColor;
+  onChange: (newValue: Partial<HsvaColor>) => void;
+}
+
+const Value: React.FC<ValueProps> = React.memo(
+  ({ className, color, onChange }) => {
+    const handleMove = (interaction: Interaction) => {
+      onChange({ v: 100 * interaction.left });
+    };
+
+    const handleKey = (offset: Interaction) => {
+      onChange({
+        v: clamp(color.v + offset.left * 100, 0, 100),
+      });
+    };
+
+    const nodeClassName = classes(['react-colorful__value', className]);
+
+    const background = useMemo(
+      () =>
+        `linear-gradient(to right, ${hsvaToHslString({
+          h: color.h,
+          s: color.s,
+          v: 0,
+          a: 1,
+        })}, ${hsvaToHslString({ h: color.h, s: color.s, v: 100, a: 1 })})`,
+      [color],
+    );
+
+    return (
+      <div className={nodeClassName}>
+        <Interactive
+          style={{
+            background,
+          }}
+          onMove={handleMove}
+          onKey={handleKey}
+          aria-label="Value"
+          aria-valuenow={Math.round(color.v)}
+          aria-valuemax={100}
+          aria-valuemin={0}
+        >
+          <Pointer
+            className="react-colorful__value-pointer"
+            left={color.v / 100}
+            color={hsvaToHslString(color)}
+          />
+        </Interactive>
+      </div>
+    );
+  },
+);
+
+interface RGBSliderProps {
+  className?: string;
+  color: HsvaColor;
+  onChange: (newValue: HsvaColor) => void;
+  target: 'r' | 'g' | 'b';
+}
+
+const RGBSlider: React.FC<RGBSliderProps> = React.memo(
+  ({ className, color, onChange, target }) => {
+    const rgb = hsvaToRgba(color);
+
+    const setNewTarget = (value: number) => {
+      const newRgb = { ...rgb, [target]: value };
+      onChange(rgbaToHsva(newRgb));
+    };
+
+    const handleMove = (interaction: Interaction) => {
+      setNewTarget(255 * interaction.left);
+    };
+
+    const handleKey = (offset: Interaction) => {
+      setNewTarget(clamp(rgb[target] + offset.left * 255, 0, 255));
+    };
+
+    const nodeClassName = classes([`react-colorful__${target}`, className]);
+
+    const selected =
+      target === 'r'
+        ? `rgb(${Math.round(rgb.r)},0,0)`
+        : target === 'g'
+          ? `rgb(0,${Math.round(rgb.g)},0)`
+          : `rgb(0,0,${Math.round(rgb.b)})`;
```

Comment:
Simplify this instead of making a tree of ternaries, please

Reply:
I replaced the nested ternaries with a simple object-based lookup.